docs/conf.py

```diff
@@ -156,3 +167,38 @@
 
 # -- Linkcheck options ------------------
 linkcheck_anchors_ignore = ["/#!"]
+
+# -- Build the latest JS for local preview -----------------------------
+from subprocess import run
+from pathlib import Path
+import os
+
+
+print('*************************** RUNNING CONF.PY ***************************')
+
+# -- Run JS Build if needed ------------------------
+# Note: this will be a one off run on RTD but may be run mulitple times locally
+# during development and testing
+path_root = Path(__file__).parent.parent
+
+if os.environ.get('READ_THE_DOCS'):
```

Comment:
What's the reason for having two different workflows on RTD vs. locally? I'm wondering if it'd be possible to just have one check for whether the `_static/lib` folder exists rather than making it build environment-specific 🤔

Reply:
That's a good point @choldgraf. 

I'm seeing the RTD process as separate mostly due to the unique setup it needs i.e. because of the local installs of `yarn` and `jsdoc` which are typically going to be global installs in someone's local development environment. We don't want two versions of these kicking around locally, or to add them to the developer's path.

I have separated that environment setup from the build commands, where the latter are now a run based on a check for the `_static/lib` in both environments as you suggested.

Note in the course of this I've picked up on some more changes that were needed to properly support the "use local build" both locally and on RTD. see changes to `Makefile` and `webpack.config.js`.